packages/ckb-sdk-core/src/signWitness.ts

```diff
@@ -0,0 +1,12 @@
+import { blake2b, hexToBytes, bytesToHex, PERSONAL } from '@nervosnetwork/ckb-sdk-utils'
+import Address from './address'
+
+export default (addrObj: Address, transactionHash: string, witness: string = '') => {
+  const s = blake2b(32, null, null, PERSONAL)
+  const witnessBytes = hexToBytes(witness)
+  s.update(hexToBytes(transactionHash))
+  s.update(witnessBytes)
+  const message = `0x${s.digest('hex')}`
+  const data = [...hexToBytes(addrObj.signRecoverable(message)), ...witnessBytes]
```

Comment:
Why `signRecoverable ` function is in `Address` module ?

Reply:
In fact, the `Address` class derives from the `ECPairs` class, which is used to generate key pairs, thus the address object is able to sign messages.

Reply:
OK. I just think `Address` and `ECPairs` are different module, and `signRecoverable ` can be in `ECPairs` but not `Address`.

Reply:
Agree. I have been wanting to propose changing `Address`'s base type for long time. It's not great to let developer know that address is an `ECPairs`.

Reply:
Please plan this in another feature branch.

Reply:
Gotcha